scrapers/codechef.py

```diff
@@ -1,57 +1,117 @@
+
+        if contest_code[0:4]=="COOK":
+            easy_points=100000 # Initial value set to points per problem
+
+        for Division in divisions: # Build the scraped scoreboard
+            divisions[Division].problems = get_problems(f"{site_url}/{contest_code}{Division}")
+            divisions[Division].scraped_scoreboard = get_rankings(f"{scoreboard_base_url}/{contest_code}{Division}{scoreboard_filter_query}", contest_code)
+
+        easy_points=len(divisions['B'].problems-divisions['A'].problems)*easy_points # Points to add to div-A participants
+
+        for i in range(len(divisions['A'].scraped_scoreboard)): # Add easy points to all div-A participants
+            divisions['A'].scraped_scoreboard[i]=divisions['A'].scraped_scoreboard[i][0],divisions['A'].scraped_scoreboard[i][1]+easy_points
+            
+        final_scoreboard.extend(divisions['A'].scraped_scoreboard)
+        final_scoreboard.extend(divisions['B'].scraped_scoreboard) # Consolidate scoreboard
+        final_scoreboard = sorted(final_scoreboard,key=lambda x:x[1],reverse=True) # Sort list in desc. order based on points. (username, points)
+
+        if contest_code[0:5] == 'LTIME':
+            month, year = find_month_year(6, contest_code[5:], 13)
+            contest_name = f'codechef-{month}-lunchtime-{year}.in'
+        elif contest_code[0:4]=="COOK":
+            month, year = find_month_year(8, contest_code[4:], 10)
+            contest_name = f'codechef-{month}-cookoff-{year}.in'
+        else:
+            month = contest_code[:-2].lower()
+            contest_name = f'codechef-{month}-long-{contest_code[-2:]}.in'
+
+        if final_scoreboard: # If scoreboard's not empty
+            rank_list = []
+            contest_ranks_file = f'database/contest_ranks/{contest_name}'
+            if contest_code[0:5]=="LTIME" or contest_code[0:4]=="COOK": 
+                rank_list = [x[0] for x in final_scoreboard]
+            else: # Shared ranking possible for long contests.
+                last_score = final_scoreboard[0][1]
+                shared_rank = []
+                for user in final_scoreboard:
+                    if user[1] == last_score:
+                        shared_rank.append(user[0])
+                        if final_scoreboard.index(user) == len(final_scoreboard) - 1:
+                            rank_list.append(' '.join(shared_rank))
+                    else:
+                        rank_list.append(' '.join(shared_rank))
+                        shared_rank = []
+                        last_score = user[1]
+                        shared_rank.append(user[0])
+            return contest_ranks_file, rank_list
+        #info(f'Leaderboard written to {contest_ranks_file}')
         print()
-        lowest_so_far = rank
-    print(handle, end=' ')
+    print()
+    driver.close()
+
+def find_month_year(offSet, ID, start):
```

Comment:
Some documentation required here, and move this to the main file too

Reply:
this is used only for the codechef scraper though